### PR TITLE
Add Contractors team

### LINKED ISSUE #290: Create contractor sub team and reduce team membership duplication
# Proposal
Reduce number of duplicate entries of individuals across teams to make managing permissions more straightforward

- Reduce Yoshi direct membership to a minimum number of people. Rely on sub teams to manage this
- All repositories contain the `yoshi` team, so having all people in each language team is redundant
- Create `yoshi-contractors` to migrate contractor team members off of parent `yoshi` team.
- Remove contractors from individual language teams. The rights given via yoshi are sufficient.
- Give Yoshi team write access to all repositories (some currently have read)
- Enable master branch protection on all repositories
- Create `CODEOWNERS` files for all repositories

### PR DESCRIPTION
**Pull request by crwilcox**
Fixes #290

### CODE REVIEW

**Comment by JustinBeckwith** [comment]
I like where your heads at, but as of right now, I don't think this would work.  We use the `yoshi-<language>` groups as a way to decide who gets write access to which repos in many languages (nodejs I know for sure).  When we have no promises of master branch protection or owner files, I get nervous giving anyone write to the repo.

**Comment by crwilcox** [comment]
@JustinBeckwith, I can't see the node permissions but we have, in the repos I have admin to, mirrored permission level between `yoshi-language` and `yoshi`. Since `yoshi` includes the language teams, is redundant. Is it the case that you have differentiated this in node?

**Comment by JustinBeckwith** [comment]
Yeah, for better or for worse, we did.  We have `yoshi` with read level access, `yoshi-nodejs` with write, and `yoshi-admins` with admin.

As we've discussed though, I'm ok with opening things up *if* we enforce master branch protection and CODEOWNERs files.

File: users.json [hunk 1]
```diff
@@ -390,6 +330,40 @@
         "googleapis/google-cloud-cpp",
         "googleapis/google-cloud-cpp-spanner"
       ]
+    },
+    {
+      "team": "yoshi-contractors",
+      "users": [
+        "abhinav-qlogic",
```
Comment (crwilcox) [inline users.json]: @danoscarmike this list should include all of the contractors. Are any missing to your knowledge?

**Comment by crwilcox** [comment]
@JustinBeckwith 100% agree. We should have master branch protection on anyway. And CODEOWNERS seems a better way to cross-section PA and Language expertise than a language specific list across the board.

I seem to recall that Ruby might have a process that wouldn't mesh with master branch protection? @blowmage @quartzmo @TheRoyalTnetennba @dazuma ?

**Comment by blowmage** [comment]
> I seem to recall that Ruby might have a process that wouldn't mesh with master branch protection?

I think Ruby is far enough into adoption of the yoshi toolset that we no longer need master branch protection. I believe this is already enabled for googleapis/google-cloud-ruby.

**Comment by crwilcox** [comment]
Marking DNM, as this is sort of a proof of concept.  I intend to write a formal design for this and how we should reconfigure our permissions and such.

File: users.json (hunk 1 above)
Comment (danoscarmike) [reply users.json]: Added one additional user

**Comment by JustinBeckwith** [comment]
Pretty sure we decided not to do that :)

**Pull request closed without merging**